From: Jory A. Pratt <anarchy@gentoo.org>

Note: the first occurrence of :space: is [:space:] (with a single pair
of []), where it should be [[:space:]] (with two pairs of []).

This causes the sed command to fail (with the error message quoted in
my original report), driving to failure the whole configure step.

https://bugzilla.mozilla.org/show_bug.cgi?id=1329252

diff --git a/build/autoconf/icu.m4 b/build/autoconf/icu.m4
--- a/build/autoconf/icu.m4
+++ b/build/autoconf/icu.m4
@@ -70,7 +70,7 @@
         fi
     fi
 
-    version=`sed -n 's/^[[:space:]]*#[[:space:]]*define[[:space:]][[:space:]]*U_ICU_VERSION_MAJOR_NUM[[:space:]][[:space:]]*\([0-9][0-9]*\)[[:space:]]*$/\1/p' "$icudir/common/unicode/uvernum.h"`
+    version=`sed -n 's/^[[[:space:]]]*#[[:space:]]*define[[:space:]][[:space:]]*U_ICU_VERSION_MAJOR_NUM[[:space:]][[:space:]]*\([0-9][0-9]*\)[[:space:]]*$/\1/p' "$icudir/common/unicode/uvernum.h"`
     if test x"$version" = x; then
        AC_MSG_ERROR([cannot determine icu version number from uvernum.h header file $lineno])
     fi